server/viame_server/__init__.py

```diff
@@ -32,11 +32,21 @@ def load_pipelines():
 
     allowed = r"^detector_.+|tracker_.+"
     disallowed = r".*local.*|detector_svm_models.pipe|tracker_svm_models.pipe"
-    return [
+    pipelist = [
         path.name
         for path in pipeline_path.glob("./*.pipe")
         if re.match(allowed, path.name) and not re.match(disallowed, path.name)
     ]
+    pipedict = {
+        'detector': {'pipes': [], 'description': '',},
+        'tracker': {'pipes': [], 'description': '',},
+    }
+    for pipe in pipelist:
+        nameparts = pipe.replace('.pipe', '').split('_')
+        name = ' '.join(nameparts[1:])
+        type = nameparts[0]
+        pipedict[type]['pipes'].append({'name': name, 'type': type, 'pipe': pipe})
+    return pipedict
```

Comment:
A few things here in my suggestion:

1. Double quotes to match black style
2. Changing the `type` variable to `pipe_type`, since `type` is a keyword in python
3. Use a shorthand for splitting the `nameparts` array into the first and remaining elements
4. Generalizing this to any "type", since we will probably be adding another once #112 is finalized. Since your approach already grabbed the first part of the pipeline name and used it as the dict key, this just takes full advantage of that.

Lmk what you think


```suggestion
    pipedict = {}
    for pipe in pipelist:
        pipe_type, *nameparts = pipe.replace(".pipe", "").split("_")
        pipe_info = {"name": " ".join(nameparts), "type": pipe_type, "pipe": pipe}

        if pipe_type in pipedict:
            pipedict[pipe_type]["pipes"].append(pipe_info)
        else:
            pipedict[pipe_type] = {"pipes": [pipe_info], "description": ""}

    return pipedict
```